### PR TITLE
worker bridge - idempotency key

### PR DESCRIPTION
**Pull request by justcoon**

### CODE REVIEW

**Review by afsalthaj — APPROVED** [review]

**Comment by jdegoes** [comment]
@justcoon The worker bridge needs to pull idempotency keys from the HTTP header. 

See [here](https://datatracker.ietf.org/doc/draft-ietf-httpapi-idempotency-key-header/)

cc @vigoo

**Comment by vigoo** [comment]
Currently you can set it using Rig any way you want (for example by forwarding that or any other header) but it's not happening automatically

**Comment by afsalthaj** [comment]
Setting the idempotency key is per binding, and this part was already done in @vigoo 's big PR

**Comment by vigoo** [comment]
I think @justcoon 's latest commits also implement https://github.com/golemcloud/golem/issues/324

File: golem-worker-service-base/src/worker_bridge_execution/mod.rs [hunk 1]
```diff
@@ -77,6 +78,28 @@ impl WorkerRequest {
...
+            let idempotency_key = match idempotency_key_value {
+                TypeAnnotatedValue::Str(value) => value,
+                _ => return Err("Idempotency Key is not a string".to_string()),
+            };
+
+            Some(IdempotencyKey::new(idempotency_key))
+        } else {
+            resolved_route
+                .headers
+                .get("idempotency-key")
```
Comment (vigoo) [inline golem-worker-service-base/src/worker_bridge_execution/mod.rs:98]: Is this working case-insensitively?

**Review by vigoo — APPROVED** [review]

File: golem-worker-service-base/src/worker_bridge_execution/mod.rs (hunk 1 above)
Comment (justcoon) [reply golem-worker-service-base/src/worker_bridge_execution/mod.rs:98]: headers should not be case sensitive, i added test

**Pull request merged**